### DISCUSSION TITLE
Tooling for doing data "imports" from other social networks

### DISCUSSION THREAD

**Discussion by pfrazee** [bluesky-social/atproto#1152] [General Discussion]
Some folks have been wanting to build importers of posts and data from other networks.

There are a lot of tricks to that but a very _basic_ import has some blockers that we'd need to address:

1. The timestamps are going to be based on index time, not create time. This can be fixed by using a "blended" timestamp where, depending on the view, we use either created or indexed times.  Eg on a reverse-chron timeline you choose the lower of the two.This technique only works contextually though, and may be purely a client design choice.
2. The posts will be treated as normal posts by the indexer, meaning they'll blast/swap the followers' timelines as new objects. There would need to be some way to signal that the post is an import and should somehow get indexed "silently".

This is somewhat challenging and I don't think we can prioritize it now. Also these are only 2 issues that occurred to me at first glance, there may be more issues.

**Comment by ianklatzco** [comment]
I'm curious if the WIP importRepo would still cause posts to render based on indexedAt.

Also curious where the code is that decides whether to use indexedAt vs createdAt -- in the UI? or in the getTimeline?

![image](https://user-images.githubusercontent.com/1487413/231385388-f5c5a3e8-18a9-4edb-9d18-212cd9396b5b.png)

**Comment by kcchu** [comment]
I would like to spend some time on AT Protocol's code. I am interested in working on this task. Are there anyone already started working on this or some prior work?
